### ISSUE TITLE
We shouldn't require a full `javax.cache.Cache` implementation

### ISSUE THREAD

**Issue by metacosm** [java-operator-sdk/java-operator-sdk#756]
Adding that dependency makes it more complex to consume the API as it requires a complex cache implementation when we only need a subset… In particular, using a simpler Cache interface could potentially allow us to make `InformerEventSource` a `CachingEventSource` and could allow us to propagate changes to the informer's caches directly from the SDK instead of having to wait for them to come from the API server.

**Comment by csviri** [comment]
Agree. For now will remove it from the new event source implementations. Original idea was, that there caches could have eviction, limits and other features that we could use. But at the end even the logic in `CachingEventSource` kinda assumes that the values are caches forever.
